### ISSUE TITLE
Staging rows with a null value causes inconvenience

### ISSUE THREAD

**Issue by li6in9muyou** [li6in9muyou/SwingDbCrudApp#2]
# To reproduce
1. create a table with a column of type `Date`.
2. create a row where this `Date` attribute is `null`.
3. stage that row.
4. change primary key attribute of staged row so that it can be insert back
5. click commit button.

# Expected behaviour
After step 5, a row is inserted into the table with all identical attributes with the previous row except for primary key.

# Actual behaviour
After step 3, staging area shows this null `Date` typed attribute as string "null".
After step 5, this app reports error saying `For input string: "null"`.

**Comment by li6in9muyou** [comment]
As a temporary fix, when staging a row into `singleLineInsert` staging area, it converts all Java `null` value to empty string instead of `"null"`.

**Comment by li6in9muyou** [comment]
This approach is not going to work. Before sending SQL statement back to database, the JDBC client must distinguish "literal empty strings" between "empty strings representing a null value". We will have to develop so kind of escape code for this purpose. 
I am not comfortable with such strong coupling between JTable representation and JDBC client code. This part of the system needs through refactoring and I will open a new issue to track that.
